core/src/main/scala/codesearch/core/index/Sources.scala

```diff
@@ -39,80 +37,39 @@ trait Sources[VTable <: DefaultTable] {
     }
   }
 
-
   def update(): Future[Int] = {
     val lastVersions = indexAPI.getLastVersions.mapValues(_.verString)
 
-    val futureAction = indexAPI.verNames().flatMap { packages =>
-      val packagesMap = Map(packages: _*)
-
-      Future.sequence(lastVersions.filter {
-        case (packageName, currentVersion) =>
-          !packagesMap.get(packageName).contains(currentVersion)
-      }.map {
-        case (packageName, currentVersion) =>
-          downloadSources(packageName, currentVersion)
-      })
-    }.map(_.sum)
+    val futureAction = indexAPI
+      .verNames()
+      .flatMap { packages =>
+        val packagesMap = Map(packages: _*)
+
+        Future.sequence(lastVersions.filter {
+          case (packageName, currentVersion) =>
+            !packagesMap.get(packageName).contains(currentVersion)
```

Comment:
It is slightly more "scalish" to use `filterNot` without `!` inside.

Reply:
Thnx, it is all will be refactored.